core/src/main/java/org/apache/iceberg/util/SnapshotUtil.java

```diff
@@ -65,12 +65,16 @@ public static boolean ancestorOf(Table table, long snapshotId, long ancestorSnap
   }
 
   /**
-   * Traverses the history of the table's current snapshot and finds the oldest Snapshot.
-   * @return null if there is no current snapshot in the table, else the oldest Snapshot.
+   * Traverses the history of the table's current snapshot and finds the oldest Snapshot after or equal to the timestamp in milliseconds.
+   * @return null if there is no current snapshot in the table, else the oldest Snapshot after or equal to the timestamp in milliseconds.
    */
-  public static Snapshot oldestSnapshot(Table table) {
+  public static Snapshot oldestSnapshot(Table table, long timestampMillis) {
```

Comment:
Once we confirm this is definitely the same semantics of `as-of-timestamp`, can we just rename this to be `snapshotAsOfTimstamp`?

Reply:
EDIT - Looking further at the clarification, I see I was mistaken.

With the updated doc comment and the change in Boolean logic, this is fine. No need to rename. Or if considering renaming, I'd do `snapshotOlderThan(Table table, long timestamp)` or something.

Reply:
I think the original naming oldestSnapshot makes more sense, since snapshotOlderThan could be any snapshot after the timestamp. But oldestSnapshot clarify the one just after the timestamp, and the documentation also provides this information

Reply:
Yeah I don't think renaming is needed either. It's just still somewhat odd to me as the notion of CDC isn't really baked in.

I have a few ideas for that though that I'm still thinking about.